### PR TITLE
only accept paths in dusty reposs override that are accurate wrt NFS

### PR DESCRIPTION
**Pull request by paetling**
review @thieman  solves https://github.com/gamechanger/dusty/issues/547

### CODE REVIEW

File: dusty/commands/repos.py [hunk 1]
```diff
@@ -21,10 +21,19 @@ def list_repos():
...
+        if path_element not in os.listdir(recreated_path):
+            return False
+        recreated_path = "{}{}/".format(recreated_path, path_element)
+    return True
+
 @daemon_command
 def override_repo(repo_name, source_path):
     repo = Repo.resolve(get_all_repos(), repo_name)
-    if not os.path.exists(source_path):
+    if not nfs_path_exists(source_path):
```
Comment (thieman) [inline dusty/commands/repos.py]: Need a comment here explaining why we're doing this weird stuff

File: dusty/commands/repos.py [hunk 2]
```diff
@@ -21,10 +21,19 @@ def list_repos():
                        repo.override_path if repo.is_overridden else ''])
     log_to_client(table.get_string(sortby='Short Name'))
 
+def nfs_path_exists(path):
```
Comment (thieman) [inline dusty/commands/repos.py]: This function seems insane if you don't know why it's here, docstring would be great

File: docs/changelog.md [hunk 3]
```diff
@@ -11,6 +11,7 @@
     * Host forwarding now supports multiple forwarding rules per app to the same container port for HTTP forwarding
     * Fixed a bug where intermediate containers were not removed when creating testing images
     * Revived a previous networking fix that was removed after Dusty moved to Docker Machine. This fix improves external networking performance by up to 50% for large file transfers.
+    * Fixed a bug which allowed a directory to be written to in HFS but did not allow that same directory to be pulled using NFS.
```
Comment (thieman) [inline docs/changelog.md]: Would word this more generally, just mention fixing some case-sensitivity bug or something

File: dusty/commands/repos.py [hunk 4]
```diff
@@ -21,10 +21,19 @@ def list_repos():
...
+    split_path = path.lstrip('/').split('/')
+    recreated_path = '/'
+    for path_element in split_path:
+        if path_element not in os.listdir(recreated_path):
```
Comment (thieman) [inline dusty/commands/repos.py]: This does in fact return case-sensitive results, right? Any way to do a unit test with it against actual files on an HFS filesystem?

Comment (paetling) [inline dusty/commands/repos.py]: can do

**Pull request merged**